project-base/storefront/components/providers/CookiesStoreProvider.tsx

```diff
@@ -0,0 +1,15 @@
+import { createContext, useRef } from 'react';
+import { CookiesStore, createCookiesStore } from 'store/useCookiesStore';
+import { type StoreApi } from 'zustand';
+
+export const CookiesStoreContext = createContext<StoreApi<CookiesStore> | null>(null);
+
+export const CookiesStoreProvider: FC = ({ children }) => {
+    const storeRef = useRef<StoreApi<CookiesStore>>();
+
+    if (!storeRef.current) {
+        storeRef.current = createCookiesStore();
+    }
```

Comment:
Isn't this the exact same code, but more clear? 

```suggestion
    const storeRef = useRef<StoreApi<CookiesStore>>(createCookiesStore());
```

Reply:
Good idea! Yes, it is cleaner for sure.